### PR TITLE
Clean editpart children

### PR DESCRIPTION
**Pull request by azoitl**
In order to address #155 this PR reattempts to turn the children list and all its handling int java generics. For most users this PR should not provide any issues.

@Destrolaric what do you think?

@pcdavid I tried hard but I break 2 lines in GMF runtime. I would submit a PR to fix it. How and when would you like to have it?

### CODE REVIEW

**Comment by pcdavid** [comment]
> @pcdavid I tried hard but I break 2 lines in GMF runtime. I would submit a PR to fix it. How and when would you like to have it?

I'll try to have a look this week, but don't know when.

File: org.eclipse.gef.examples.text/src/org/eclipse/gef/examples/text/SelectionRange.java [hunk 1]
```diff
@@ -86,78 +86,75 @@ public SelectionRange(TextEditPart begin, int bo, TextEditPart end, int eo) {
...
-			depthFirstTraversal((EditPart) children.get(i), result);
+		for (int i = start; i < children.indexOf(right); i++)
+			depthFirstTraversal(children.get(i), result);
 
 		result.addAll(rightSide);
 		return result;
 	}
 
-	private List findNodesBetweenInclusive(EditPart left, EditPart right) {
+	private static List<EditPart> findNodesBetweenInclusive(EditPart left, EditPart right) {
```
Comment (Destrolaric) [inline org.eclipse.gef.examples.text/src/org/eclipse/gef/examples/text/SelectionRange.java:148]: Are you sure transferring this method to a static context will not break anything?

File: org.eclipse.gef.examples.text/src/org/eclipse/gef/examples/text/TextEditor.java [hunk 2]
```diff
@@ -78,6 +78,35 @@ public class TextEditor extends GraphicalEditor {
...
+		@Override
+		public EditPart createEditPart(EditPart context, Object model) {
+			if (model instanceof Container cont) {
+				switch (cont.getType()) {
+				case Container.TYPE_ROOT:
+					return new DocumentPart(cont);
+				case Container.TYPE_IMPORT_DECLARATIONS:
+					return new ImportsPart(cont);
+
+				case Container.TYPE_COMMENT, Container.TYPE_PARAGRAPH:
```
Comment (Destrolaric) [inline org.eclipse.gef.examples.text/src/org/eclipse/gef/examples/text/TextEditor.java:91]: Minor formatting issue

**Review by Destrolaric — APPROVED** [review]
Looks good, some minor questions.

File: org.eclipse.gef.examples.text/src/org/eclipse/gef/examples/text/SelectionRange.java (hunk 1 above)
Comment (azoitl) [reply org.eclipse.gef.examples.text/src/org/eclipse/gef/examples/text/SelectionRange.java:148]: As it is private and not accessing any fields and it was suggested by the Eclipse Java compiler I'm pretty sure.

File: org.eclipse.gef.examples.text/src/org/eclipse/gef/examples/text/TextEditor.java (hunk 2 above)
Comment (azoitl) [reply org.eclipse.gef.examples.text/src/org/eclipse/gef/examples/text/TextEditor.java:91]: How would you format this? I thought the new case statement with commas between cases is formated on one line.

File: org.eclipse.gef.examples.text/src/org/eclipse/gef/examples/text/SelectionRange.java (hunk 1 above)
Comment (azoitl) [reply org.eclipse.gef.examples.text/src/org/eclipse/gef/examples/text/SelectionRange.java:148]: As this method is private, not accessing any fields, and as it was suggested by the Eclipse Java compiler I'm pretty sure. But if you are not feeling comfortable I'm happy to change back.

**Comment by azoitl** [comment]
> > @pcdavid I tried hard but I break 2 lines in GMF runtime. I would submit a PR to fix it. How and when would you like to have it?
> 
> I'll try to have a look this week, but don't know when.

Please take your time. I have the fix for GMF ready just in case.

**Comment by azoitl** [comment]
@pcdavid I just saw your commit. So I think your are fine with merging it? Please note that I'm happy to perform such API change commits for GMF runtime as well.

**Comment by pcdavid** [comment]
Let me check again, I'm not sure that patch compiled properly.

**Comment by pcdavid** [comment]
I've merged the GMF part, you can go ahead.

**Comment by azoitl** [comment]
> I've merged the GMF part, you can go ahead.

thx. Then I will do. Have a few more patches in the pipeline that I would love to have in RC1 to have early tests. For that I also changed a few of my test setups to catch things we missed in 3.16.

**Pull request merged**